cpp/examples/mid_level_writer_example.cc

```diff
@@ -0,0 +1,181 @@
+/** Copyright 2022 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include <iostream>
+
+#include "arrow/api.h"
+#include "arrow/csv/api.h"
+#include "arrow/filesystem/api.h"
+#include "parquet/arrow/reader.h"
+
+#include "./config.h"
+#include "gar/writer/arrow_chunk_writer.h"
+
+std::shared_ptr<arrow::Table> read_csv_file_and_return_table(
```

Comment:
Like `VerticesBuilder` above, maybe just construct a simple table is OK?

Reply:
> Like `VerticesBuilder` above, maybe just construct a simple table is OK?

Yes, I agree. I will modify it to construct a simple table using hardcoded values here.